### ISSUE TITLE
PointerMoveEventFilter.onEnter not firing during mouse wheel scrolling

### ISSUE THREAD

**Issue by Inego** [JetBrains/compose-jb#523]
Hovering the mouse pointer over list items correctly updates the "HOVERED:" text.
Scrolling the list with the mouse wheel while the pointer is over the list doesn't (although intuitively it should).

```kotlin
import androidx.compose.desktop.Window
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.lazy.rememberLazyListState
import androidx.compose.material.Text
import androidx.compose.runtime.mutableStateOf
import androidx.compose.ui.Modifier
import androidx.compose.ui.input.pointer.pointerMoveFilter


fun main() {

    val lines = (0..1000).map { "Line $it" }

    val hovered = mutableStateOf("")

    Window {
        Column {
            Text("HOVERED: ${hovered.value}")

            val lazyListState = rememberLazyListState()

            LazyColumn(state = lazyListState) {
                items(lines) {
                    Text(it, modifier = Modifier.pointerMoveFilter {
                        hovered.value = it
                        false
                    })
                }
            }
        }
    }
}
```

**Comment by akurasov** [comment]
if mouse is not moved, then mouse move event shouldn't be triggered. I believe you should listen to scroll events and detect item under mouse cursor based on it. The situation is not unique - e.g. https://stackoverflow.com/questions/8209031/how-to-detect-that-the-mouse-is-over-an-element-during-a-scroll

**Comment by igordmn** [comment]
>  listen to scroll events 
The problem is not only with scroll, but with any layout change (for example, hide/show element).

If the user writes a common widget that should be hoverable, it should be easy to write the hover logic that works in any case.

There are two options to solve the issue:
1. don't change low-level logic, write high-level logic `Modifier.onHover(action: (hovered: Boolean) -> Unit)` which will take into account layout changes (via `onGloballyPositioned`)
2. change semantics of low-level onMove/onEnter/onExit - we will trigger them when the relative position will be changed, not on a mouse event.

**Comment by akurasov** [comment]
There was a request for hoverable modifier. I would say it is more user(=developer)-friendly to have it, then have a lot of low-level functions making everyone to develop hoverable themselves

**Comment by igordmn** [comment]
Should be fixed in 1.0.0

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5472">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
